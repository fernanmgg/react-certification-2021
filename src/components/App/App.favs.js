export default [
  {
    name: 'Wizeline',
    videos: [
      {
        id: {
          videoId: 'nmXMgqjQzls',
        },
        snippet: {
          title: 'Video Tour | Welcome to Wizeline Guadalajara',
          description:
            'Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...',
          thumbnails: {
            medium: {
              url: 'https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg',
            },
          },
        },
      },
      {
        id: {
          videoId: 'HYyRZiwBWc8',
        },
        snippet: {
          title: 'Wizeline Guadalajara | Bringing Silicon Valley to Mexico',
          description:
            'Wizeline continues to offer a Silicon Valley culture in burgeoning innovation hubs like Mexico and Vietnam. In 2018, our Guadalajara team moved into a ...',
          thumbnails: {
            medium: {
              url: 'https://i.ytimg.com/vi/HYyRZiwBWc8/mqdefault.jpg',
            },
          },
        },
      },
      {
        id: {
          videoId: 'NkZkhISXZ_Q',
        },
        snippet: {
          title: 'Wizeline Hiring Process Overview',
          description:
            'Have you ever wondered what is Wizeline looking for in a Software Engineer and how our hiring process looks? Look no further! In this video, we explain ...',
          thumbnails: {
            medium: {
              url: 'https://i.ytimg.com/vi/NkZkhISXZ_Q/mqdefault.jpg',
            },
          },
        },
      },
      {
        id: {
          videoId: 'cjO2fJy8asM',
        },
        snippet: {
          title: 'A Day in the Life of an Engineering Manager at Wizeline',
          description:
            "Fernando Espinoza shares his experience working as an engineering manager at Wizeline and mentoring other engineers. Learn about Fernando's passions ...",
          thumbnails: {
            medium: {
              url: 'https://i.ytimg.com/vi/cjO2fJy8asM/mqdefault.jpg',
            },
          },
        },
      },
      {
        id: {
          videoId: 'bkX4bBVe9R8',
        },
        snippet: {
          title: 'Acing the Wizeline Technical Assessment',
          description:
            'Check out this video to learn more about the technical assessment at Wizeline and prepare yourself to succeed in your interviewing process.',
          thumbnails: {
            medium: {
              url: 'https://i.ytimg.com/vi/bkX4bBVe9R8/mqdefault.jpg',
            },
          },
        },
      },
      {
        id: {
          videoId: 'WDtNW4vjBfg',
        },
        snippet: {
          title: 'A Day in the Life of a Senior Software Engineer at Wizeline',
          description:
            'Meet Jakob Renpening, Senior Software Engineer at Wizeline. Jakob shares with us some of his passions and his day-to-day activities. He also explains why he ...',
          thumbnails: {
            medium: {
              url: 'https://i.ytimg.com/vi/WDtNW4vjBfg/mqdefault.jpg',
            },
          },
        },
      },
      {
        id: {
          videoId: '7PtYNO6g7eI',
        },
        snippet: {
          title: 'We Are Wizeline',
          description:
            'Engineering a better tomorrow. Wizeline is a global software development company that helps its clients solve their biggest challenges with design and ...',
          thumbnails: {
            medium: {
              url: 'https://i.ytimg.com/vi/7PtYNO6g7eI/mqdefault.jpg',
            },
          },
        },
      },
      {
        id: {
          videoId: 'Po3VwR_NNGk',
        },
        snippet: {
          title: 'Wizeline hace sentir a empleados como en casa',
          description:
            'En el 2014, Bismarck fundó Wizeline, compañía tecnológica que trabaja con los corporativos ofreciendo una plataforma para que desarrollen software de forma ...',
          thumbnails: {
            medium: {
              url: 'https://i.ytimg.com/vi/Po3VwR_NNGk/mqdefault.jpg',
            },
          },
        },
      },
      {
        id: {
          videoId: 'iszLGKZeFsk',
        },
        snippet: {
          title: 'Wizeline&#39;s Guadalajara Campus',
          description:
            "Wizeline's state-of-the-art campus in Guadalajara, Mexico, has the capacity for 700 employees. The facility is designed to foster collaboration with remote teams, ...",
          thumbnails: {
            medium: {
              url: 'https://i.ytimg.com/vi/iszLGKZeFsk/mqdefault.jpg',
            },
          },
        },
      },
      {
        id: {
          videoId: '3KVFmT-Tp2w',
        },
        snippet: {
          title: 'Caso de Éxito AWS: Wizeline [Spanish]',
          description:
            'Central de socios de APN - https://amzn.to/2S7tIXM Fundada en 2014, Wizeline es una compañía joven e innovadora que nació en la nube para ofrecer soporte ...',
          thumbnails: {
            medium: {
              url: 'https://i.ytimg.com/vi/3KVFmT-Tp2w/mqdefault.jpg',
            },
          },
        },
      },
      {
        id: {
          videoId: 'CHzlSGRvWPs',
        },
        snippet: {
          title: 'Wizeline',
          description:
            'El plan de Wizeline, una empresa de inteligencia artificial, para ayudar a crecer la comunidad de ciencia de datos en CDMX y todo el país, a través de cursos ...',
          thumbnails: {
            medium: {
              url: 'https://i.ytimg.com/vi/CHzlSGRvWPs/mqdefault.jpg',
            },
          },
        },
      },
      {
        id: {
          videoId: 'Hm3wVzcHjnE',
        },
        snippet: {
          title: 'Humans of Wizeline | Alonso Donath - Multimedia Producer',
          description:
            "Meet Alonso Donath, Multimedia Producer at Wizeline. He joined us in 2016 and has been building the company's video brand ever since. When he's not ...",
          thumbnails: {
            medium: {
              url: 'https://i.ytimg.com/vi/Hm3wVzcHjnE/mqdefault.jpg',
            },
          },
        },
      },
    ],
  },
];
