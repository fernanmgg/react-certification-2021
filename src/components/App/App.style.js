import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  html {
    font-size: 1.125rem;
    line-height: 1.6;
    font-weight: 400;
    font-family: sans-serif;
    box-sizing: border-box;
    scroll-behavior: smooth;
    -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  }

  body {
    margin: 0;
    padding: 0;
    text-rendering: optimizeLegibility;
    background-image: linear-gradient(
      120deg,
      ${(props) => props.theme.backgroundColors[0]} 0,
      ${(props) => props.theme.backgroundColors[1]} 19%,
      ${(props) => props.theme.backgroundColors[2]} 42%,
      ${(props) => props.theme.backgroundColors[3]} 79%,
      ${(props) => props.theme.backgroundColors[4]} 100%
    );
    background-size: 400% 400%;
    background-position: var(--bg-position);
    transition: background-position 2s ease;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  *,
  *::before,
  *::after {
    box-sizing: inherit;
  }
`;

export { GlobalStyle };
