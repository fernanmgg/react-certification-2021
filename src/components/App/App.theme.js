const lightTheme = {
  accent: 'rgb(87, 198, 225)',
  accentTransparent: 'rgba(87, 198, 225, 0.25)',
  background: 'rgb(255, 255, 255)',
  backgroundColors: ['#c8abb2', '#57c6e1', '#9da9dc'],
  backgroundFocus: 'rgba(0, 0, 0, 0.16)',
  backgroundTransparent: 'rgba(255, 255, 255, 0.5)',
  title: 'rgba(255, 255, 255, 0.8)',
  text: 'rgba(0, 0, 0, 0.8)',
  textLight: 'rgba(0, 0, 0, 0.5)',
};

const darkTheme = {
  accent: 'rgb(29, 8, 77)',
  accentTransparent: 'rgba(29, 8, 77, 0.25)',
  background: 'rgb(64, 64, 124)',
  backgroundColors: ['#2d073c', '#1d084d', '#1a1c37'],
  backgroundFocus: 'rgba(255, 255, 255, 0.08)',
  backgroundTransparent: 'rgba(64, 64, 124, 0.5)',
  title: 'rgba(255, 255, 255, 0.8)',
  text: 'rgba(255, 255, 255, 0.8)',
  textLight: 'rgba(255, 255, 255, 0.5)',
};

export { lightTheme, darkTheme };
