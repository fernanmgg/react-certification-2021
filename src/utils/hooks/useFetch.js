import { useState, useEffect } from 'react';

const key = process.env.REACT_APP_API_KEY;

const baseUrl = 'https://www.googleapis.com/youtube/v3/search?';

export default function useFetch(search) {
  const [videos, setVideos] = useState([]);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  useEffect(() => {
    function buildURI() {
      const fields = [
        'part=snippet',
        'fields=items(id(videoId),snippet(title,description,thumbnails(medium(url))))',
        'maxResults=20',
        'type=video',
        `key=${key}`,
        `q=${search}`,
      ];
      return baseUrl.concat(
        fields.reduce((previous, current) => previous.concat(`&${current}`))
      );
    }

    const fetchData = async () => {
      setLoading(true);
      try {
        const response = await fetch(encodeURI(buildURI()));
        const json = await response.json();
        setVideos(json.items);
      } catch (e) {
        setError(true);
      }
      setLoading(false);
    };

    if (search.length > 3) fetchData();
  }, [search]);

  return { videos, loading, error };
}
