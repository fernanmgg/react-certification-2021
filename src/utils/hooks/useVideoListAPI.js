import { useState, useEffect } from 'react';

const key = process.env.REACT_APP_YOUTUBE_API_KEY;

const baseUrl = 'https://www.googleapis.com/youtube/v3/search?';

export default function useVideoListAPI(query, relatedVideos = false) {
  const [videos, setVideos] = useState([]);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  useEffect(() => {
    function buildURI() {
      const fields = [
        'part=snippet',
        'fields=items(id(videoId),snippet(title,description,thumbnails(medium(url))))',
        'maxResults=20',
        'type=video',
        `key=${key}`,
      ];
      if (!relatedVideos) fields.push(`q=${query}`);
      else fields.push(`relatedToVideoId=${query}`);
      return baseUrl.concat(
        fields.reduce((previous, current) => previous.concat(`&${current}`))
      );
    }

    const fetchData = async () => {
      setLoading(true);
      try {
        const response = await fetch(encodeURI(buildURI()));
        const json = await response.json();
        if (json.items) setVideos(json.items);
        else setError(true);
      } catch (e) {
        setError(true);
      }
      setLoading(false);
    };

    const check = !relatedVideos
      ? query.length > 0 && query.length < 64
      : /^[A-Za-z0-9_-]{11}$/.test(query);
    if (check) fetchData();
  }, [query, relatedVideos]);

  return { videos, loading, error };
}
