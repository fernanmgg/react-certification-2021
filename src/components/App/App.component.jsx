import React, { useState } from 'react';
import { ThemeProvider } from 'styled-components';

import { GlobalStyle } from './App.style';
import { lightTheme } from './App.theme';
import Header from '../Header';
import Content from '../../views/Content';
import useFetch from '../../utils/hooks/useFetch';

function App() {
  const [search, setSearch] = useState('wizeline');
  const [video, setVideo] = useState(null);
  const { videos, loading, error } = useFetch(search);

  return (
    <ThemeProvider theme={lightTheme}>
      <GlobalStyle />
      <Header search={search} setSearch={setSearch} setVideo={setVideo} />
      <Content
        videos={videos}
        loading={loading}
        error={error}
        video={video}
        setVideo={setVideo}
      />
    </ThemeProvider>
  );
}

export default App;
