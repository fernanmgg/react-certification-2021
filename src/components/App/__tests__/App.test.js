import React from 'react';
import { render, screen } from '@testing-library/react';
import user from '@testing-library/user-event';
import 'jest-styled-components';

import App from '../App.component';
import { lightTheme, darkTheme } from '../App.theme';
import * as useVideoListAPI from '../../../utils/hooks/useVideoListAPI';

useVideoListAPI.default = jest.fn(() => ({ videos: [], loading: false, error: false }));

describe('App UI tests', () => {
  test('renders header and home view, login button if there is no auth info in storage', () => {
    render(<App />);
    const header = screen.getByRole('button', { name: /login/i });
    const homeView = screen.getByText(/react bootcamp 2021/i);
    expect(header).toBeInTheDocument();
    expect(homeView).toBeInTheDocument();
    expect(useVideoListAPI.default).toHaveBeenCalled();
    expect(useVideoListAPI.default).toHaveBeenCalledWith('wizeline');
  });

  test('renders header and home view, logout button if there is auth info in storage', () => {
    const auth = { id: '123', name: 'test', avatarUrl: 'test' };
    localStorage.setItem('auth', JSON.stringify(auth));
    render(<App />);
    const header = screen.getByRole('button', { name: /logout/i });
    const homeView = screen.getByText(/react bootcamp 2021/i);
    expect(header).toBeInTheDocument();
    expect(homeView).toBeInTheDocument();
    expect(useVideoListAPI.default).toHaveBeenCalled();
    expect(useVideoListAPI.default).toHaveBeenCalledWith('wizeline');
  });

  test('theme changes when toggle is clicked', () => {
    render(<App />);
    const themeText = screen.getAllByText(/dark mode/i);
    expect(themeText[0]).toHaveStyleRule(
      'color',
      lightTheme.textLight.replace(/,\s/g, ',')
    );
    const theme = screen.getAllByRole('checkbox', { name: /theme/i });
    user.click(theme[0]);
    expect(themeText[0]).toHaveStyleRule(
      'color',
      darkTheme.textLight.replace(/,\s/g, ',')
    );
    user.click(theme[0]);
    expect(themeText[0]).toHaveStyleRule(
      'color',
      lightTheme.textLight.replace(/,\s/g, ',')
    );
  });
});
