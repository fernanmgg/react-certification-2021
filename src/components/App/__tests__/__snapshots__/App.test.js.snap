// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App UI tests renders correctly, auth info in storage 1`] = `
<DocumentFragment>
  .c0 {
  background-color: rgba(255,255,255,0.5);
  box-shadow: 0px 4px 8px 4px rgba(0,0,0,0.2);
  color: rgba(0,0,0,0.8);
}

.c1 {
  background-color: rgb(255,255,255);
  position: fixed;
  height: 100%;
  -webkit-transition: left 0.15s linear;
  transition: left 0.15s linear;
  width: 240px;
  z-index: 3;
  left: -240px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgb(255,255,255);
  border-bottom: 1px solid rgba(0,0,0,0.5);
  border-left: none;
  border-right: none;
  border-top: none;
  color: rgba(0,0,0,0.5);
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 1rem;
  height: 64px;
  padding-left: 16px;
  text-align: left;
  -webkit-transition: background-color 1s ease;
  transition: background-color 1s ease;
  width: inherit;
}

.c2:hover {
  background-color: rgba(0,0,0,0.16);
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgb(255,255,255);
  border-bottom: 1px solid rgba(0,0,0,0.5);
  border-left: none;
  border-right: none;
  border-top: none;
  color: rgba(0,0,0,0.5);
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 1rem;
  height: 64px;
  padding-left: 16px;
  text-align: left;
  -webkit-transition: background-color 1s ease;
  transition: background-color 1s ease;
  width: inherit;
}

.c3:hover {
  background-color: rgba(0,0,0,0.16);
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 64px;
  padding: 0 24px;
}

.c12 {
  background-color: transparent;
  border: 0;
  border-radius: 100%;
  cursor: pointer;
  -webkit-transition: background-color 0.25s;
  transition: background-color 0.25s;
}

.c12:hover {
  background-color: rgba(255,255,255,0.5);
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(255,255,255,0.5);
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 32px;
  position: relative;
  width: 20ch;
}

.c16 {
  background: none;
  border: none;
  border-radius: inherit;
  color: rgba(0,0,0,0.8);
  font-size: 0.9rem;
  padding: 8px 8px 8px 36px;
  outline: none;
  -webkit-transition: background-color 0.25s;
  transition: background-color 0.25s;
  width: inherit;
}

.c16:hover {
  background-color: rgba(255,255,255,0.5);
}

.c17 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c18 {
  color: rgba(0,0,0,0.5);
  font-size: 0.8rem;
  margin-right: 8px;
}

.c20 {
  background-color: rgba(255,255,255,0.5);
  background-image: url(test);
  background-size: 100%;
  border: 0;
  border-radius: 50%;
  cursor: pointer;
  margin-left: 24px;
  min-height: 32px;
  min-width: 35px;
}

.c13 {
  fill: rgba(0,0,0,0.8);
  font-size: 1.5rem;
  height: 1em;
  width: 1em;
}

.c15 {
  margin-left: 8px;
  position: absolute;
  z-index: 1;
}

.c4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c5 {
  display: inline-block;
  height: 34px;
  position: relative;
  width: 60px;
}

.c8 {
  background-color: rgba(0,0,0,0.16);
  border-radius: 34px;
  bottom: 0;
  cursor: pointer;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c19 {
  background-color: rgba(255,255,255,0.5);
  border-radius: 34px;
  bottom: 0;
  cursor: pointer;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c10 {
  background-color: rgb(255,255,255);
  border-radius: 50%;
  bottom: 4px;
  height: 26px;
  left: 4px;
  position: absolute;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  width: 26px;
}

.c6 {
  height: 0;
  opacity: 0;
  width: 0;
}

.c6:checked ~ .c7 {
  background-color: #57c6e1;
}

.c6:checked ~ .c7 > .c9 {
  -webkit-transform: translateX(26px);
  -ms-transform: translateX(26px);
  transform: translateX(26px);
}

@media (min-width:600px) {
  .c3 {
    display: none;
  }
}

@media (max-width:600px) {
  .c14 {
    width: 100%;
  }
}

@media (max-width:600px) {
  .c17 {
    display: none;
  }
}

@media (min-width:405px) {

}

@media (min-width:770px) {

}

@media (min-width:1135px) {

}

@media (min-width:1500px) {

}

@media (min-width:1865px) {

}

<header
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        aria-label="home"
        class="c2"
      >
        Home
      </button>
      <button
        class="c3"
      >
        Dark Mode
        <div
          class="c4"
        />
        <label
          class="c5"
        >
          <input
            aria-label="theme"
            class="c6"
            type="checkbox"
          />
          <span
            class="c7 c8"
          >
            <span
              class="c9 c10"
            />
          </span>
        </label>
      </button>
      <button
        aria-label="favorites"
        class="c3"
      >
        Favorites
      </button>
      <button
        aria-label="logout"
        class="c3"
      >
        Logout
      </button>
    </div>
    <div
      class="c11"
    >
      <button
        aria-label="drawer"
        class="c12"
      >
        <svg
          class="c13"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <div
        class="c14"
      >
        <svg
          class="c13 c15"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <input
          aria-label="search"
          class="c16"
          maxlength="64"
          value="wizeline"
        />
      </div>
      <div
        class="c4"
      />
      <div
        class="c17"
      >
        <span
          class="c18"
        >
          Dark Mode
        </span>
        <label
          class="c5"
        >
          <input
            aria-label="theme"
            class="c6"
            type="checkbox"
          />
          <span
            class="c7 c19"
          >
            <span
              class="c9 c10"
            />
          </span>
        </label>
        <button
          aria-label="auth"
          class="c20"
        />
      </div>
    </div>
  </header>
  .c0 {
  color: rgba(255,255,255,0.8);
  text-align: center;
}

@media (min-width:600px) {

}

@media (max-width:600px) {

}

@media (max-width:600px) {

}

@media (min-width:405px) {

}

@media (min-width:770px) {

}

@media (min-width:1135px) {

}

@media (min-width:1500px) {

}

@media (min-width:1865px) {

}

<h1
    class="c0"
  >
    React Bootcamp 2021
  </h1>
  .c0 {
  background-color: rgba(255,255,255,0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 24px 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 10px;
}

.c2 {
  color: rgba(0,0,0,0.5);
  text-align: center;
  width: 100%;
}

@media (min-width:600px) {

}

@media (max-width:600px) {

}

@media (max-width:600px) {

}

@media (min-width:405px) {
  .c1 {
    width: 385px;
  }
}

@media (min-width:770px) {
  .c1 {
    width: 750px;
  }
}

@media (min-width:1135px) {
  .c1 {
    width: 1115px;
  }
}

@media (min-width:1500px) {
  .c1 {
    width: 1480px;
  }
}

@media (min-width:1865px) {
  .c1 {
    width: 1845px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
      >
        No videos found...
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App UI tests renders correctly, no auth info in storage 1`] = `
<DocumentFragment>
  .c0 {
  background-color: rgba(255,255,255,0.5);
  box-shadow: 0px 4px 8px 4px rgba(0,0,0,0.2);
  color: rgba(0,0,0,0.8);
}

.c1 {
  background-color: rgb(255,255,255);
  position: fixed;
  height: 100%;
  -webkit-transition: left 0.15s linear;
  transition: left 0.15s linear;
  width: 240px;
  z-index: 3;
  left: -240px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgb(255,255,255);
  border-bottom: 1px solid rgba(0,0,0,0.5);
  border-left: none;
  border-right: none;
  border-top: none;
  color: rgba(0,0,0,0.5);
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 1rem;
  height: 64px;
  padding-left: 16px;
  text-align: left;
  -webkit-transition: background-color 1s ease;
  transition: background-color 1s ease;
  width: inherit;
}

.c2:hover {
  background-color: rgba(0,0,0,0.16);
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgb(255,255,255);
  border-bottom: 1px solid rgba(0,0,0,0.5);
  border-left: none;
  border-right: none;
  border-top: none;
  color: rgba(0,0,0,0.5);
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  font-size: 1rem;
  height: 64px;
  padding-left: 16px;
  text-align: left;
  -webkit-transition: background-color 1s ease;
  transition: background-color 1s ease;
  width: inherit;
}

.c3:hover {
  background-color: rgba(0,0,0,0.16);
}

.c11 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 64px;
  padding: 0 24px;
}

.c12 {
  background-color: transparent;
  border: 0;
  border-radius: 100%;
  cursor: pointer;
  -webkit-transition: background-color 0.25s;
  transition: background-color 0.25s;
}

.c12:hover {
  background-color: rgba(255,255,255,0.5);
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(255,255,255,0.5);
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: 32px;
  position: relative;
  width: 20ch;
}

.c16 {
  background: none;
  border: none;
  border-radius: inherit;
  color: rgba(0,0,0,0.8);
  font-size: 0.9rem;
  padding: 8px 8px 8px 36px;
  outline: none;
  -webkit-transition: background-color 0.25s;
  transition: background-color 0.25s;
  width: inherit;
}

.c16:hover {
  background-color: rgba(255,255,255,0.5);
}

.c17 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.c18 {
  color: rgba(0,0,0,0.5);
  font-size: 0.8rem;
  margin-right: 8px;
}

.c20 {
  background-color: rgba(255,255,255,0.5);
  background-image: none;
  background-size: 100%;
  border: 0;
  border-radius: 50%;
  cursor: pointer;
  margin-left: 24px;
  min-height: 32px;
  min-width: 35px;
}

.c13 {
  fill: rgba(0,0,0,0.8);
  font-size: 1.5rem;
  height: 1em;
  width: 1em;
}

.c15 {
  margin-left: 8px;
  position: absolute;
  z-index: 1;
}

.c4 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c5 {
  display: inline-block;
  height: 34px;
  position: relative;
  width: 60px;
}

.c8 {
  background-color: rgba(0,0,0,0.16);
  border-radius: 34px;
  bottom: 0;
  cursor: pointer;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c19 {
  background-color: rgba(255,255,255,0.5);
  border-radius: 34px;
  bottom: 0;
  cursor: pointer;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c10 {
  background-color: rgb(255,255,255);
  border-radius: 50%;
  bottom: 4px;
  height: 26px;
  left: 4px;
  position: absolute;
  -webkit-transition: 0.4s;
  transition: 0.4s;
  width: 26px;
}

.c6 {
  height: 0;
  opacity: 0;
  width: 0;
}

.c6:checked ~ .c7 {
  background-color: #57c6e1;
}

.c6:checked ~ .c7 > .c9 {
  -webkit-transform: translateX(26px);
  -ms-transform: translateX(26px);
  transform: translateX(26px);
}

@media (min-width:600px) {
  .c3 {
    display: none;
  }
}

@media (max-width:600px) {
  .c14 {
    width: 100%;
  }
}

@media (max-width:600px) {
  .c17 {
    display: none;
  }
}

@media (min-width:405px) {

}

@media (min-width:770px) {

}

@media (min-width:1135px) {

}

@media (min-width:1500px) {

}

@media (min-width:1865px) {

}

<header
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        aria-label="home"
        class="c2"
      >
        Home
      </button>
      <button
        class="c3"
      >
        Dark Mode
        <div
          class="c4"
        />
        <label
          class="c5"
        >
          <input
            aria-label="theme"
            class="c6"
            type="checkbox"
          />
          <span
            class="c7 c8"
          >
            <span
              class="c9 c10"
            />
          </span>
        </label>
      </button>
      <button
        aria-label="login"
        class="c3"
      >
        Login
      </button>
    </div>
    <div
      class="c11"
    >
      <button
        aria-label="drawer"
        class="c12"
      >
        <svg
          class="c13"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <div
        class="c14"
      >
        <svg
          class="c13 c15"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <input
          aria-label="search"
          class="c16"
          maxlength="64"
          value="wizeline"
        />
      </div>
      <div
        class="c4"
      />
      <div
        class="c17"
      >
        <span
          class="c18"
        >
          Dark Mode
        </span>
        <label
          class="c5"
        >
          <input
            aria-label="theme"
            class="c6"
            type="checkbox"
          />
          <span
            class="c7 c19"
          >
            <span
              class="c9 c10"
            />
          </span>
        </label>
        <button
          aria-label="auth"
          class="c20"
        >
          <svg
            class="c13"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </button>
      </div>
    </div>
  </header>
  .c0 {
  color: rgba(255,255,255,0.8);
  text-align: center;
}

@media (min-width:600px) {

}

@media (max-width:600px) {

}

@media (max-width:600px) {

}

@media (min-width:405px) {

}

@media (min-width:770px) {

}

@media (min-width:1135px) {

}

@media (min-width:1500px) {

}

@media (min-width:1865px) {

}

<h1
    class="c0"
  >
    React Bootcamp 2021
  </h1>
  .c0 {
  background-color: rgba(255,255,255,0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 24px 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 10px;
}

.c2 {
  color: rgba(0,0,0,0.5);
  text-align: center;
  width: 100%;
}

@media (min-width:600px) {

}

@media (max-width:600px) {

}

@media (max-width:600px) {

}

@media (min-width:405px) {
  .c1 {
    width: 385px;
  }
}

@media (min-width:770px) {
  .c1 {
    width: 750px;
  }
}

@media (min-width:1135px) {
  .c1 {
    width: 1115px;
  }
}

@media (min-width:1500px) {
  .c1 {
    width: 1480px;
  }
}

@media (min-width:1865px) {
  .c1 {
    width: 1845px;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
      >
        No videos found...
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
